### ISSUE TITLE
Media keys don't work - Ubuntu GNOME 16.10

### ISSUE THREAD

**Issue by Enigma0** [plexinc/plex-media-player#531]
Media keys are ignored. Instead you get a large icon of a cancel sign (circle with line through it).

Basically means you must have focus on the window and use keyboard shortcuts instead of having global media control.

**Comment by wm4** [comment]
Yep, no global support. I wonder if PMP grabbing the media keys globally would be good enough? Otherwise we'd have to implement the mpris API.

> Instead you get a large icon of a cancel sign (circle with line through it).

Uh what?

**Comment by Enigma0** [comment]
@wm4 One of these guys: https://imgur.com/a/oMII0

I'd be happy enough if PMP did that. Can't see myself using two media applications at once.

**Comment by JakeStanger** [comment]
I'd also really like to see this, unfortunately I don't know what I'm doing nearly enough to begin implementing them.

**Comment by Enigma0** [comment]
@wm4 pretty please?

**Comment by Enigma0** [comment]
Media keys still do nothing - not explicitly a PMP issue but PMP should claim them so they are useful.

**Comment by Enigma0** [comment]
They can be made to work within applications as per: https://askubuntu.com/questions/48393/how-to-make-the-keyboard-media-keys-to-work-with-vlc-globally

They work in VLC & SMPlayer.

**Comment by wm4** [comment]
That would require implementing MPRIS2 support. That's a not quite trivial Linux-only dbus API. This wasn't done yet, so media keys won't work globally.

**Comment by nhubbard** [comment]
@wm4 Sorry to wake an old issue, but what about using a abstraction layer like this? https://github.com/lkrasner/mpris-cxx

**Comment by nhubbard** [comment]
@wm4 Would this be helpful? If so, where would we put it in the source tree? http://biboudis.blogspot.com/2013/11/qt-mpris2-and-clementine.html

**Comment by towo** [comment]
I'd second full MPRIS integration, especially for feature parity with Windows, where PMP is integrated into the media player controls and whatnot.

**Comment by LongChair** [comment]
I'll need to investigate this, but i also found this https://wiki.qt.io/How_to_catch_audio_keys_on_Linux_in_QML_applications

**Comment by towo** [comment]
Yeah, the audio key capture thing would be the easy solution and serve to fulfill *this* bug, but I'd advocate on going for MPRIS due to intercompatibility advantages and being a generally cleaner approach.

With MPRIS, other applications can act on the fact that Plex is playing something - VoIP applications muting/pausing your playback, for example.

**Comment by JakeStanger** [comment]
MPRIS would definitely be the ideal way to go; audio key capture is a bit of a quick-fix.

**Comment by LongChair** [comment]
Although technically MPRIS seems the right thing, i am concerned by both ways :

- Adding another dep (like the nemomobile link above) doesn't look sane to me
- using dbus API within Qt brings a lot of stuff in for a few keys.

I really wish Qt would bring support for multimedia Keys in the framework , that's where it would make sense.

**Comment by TheGreatMcPain** [comment]
As a workaround to get mpris to work.  I found that if libmpv was built with cplugins enabled this will work with plex media player.

https://github.com/hoyon/mpv-mpris
